src/app/auth/components/registration/registration.componet.ts

```diff
@@ -0,0 +1,181 @@
+import { FormatDataService } from '../../../shared/services/format-date.service';
+import { SnackbarService } from '../../../shared/services/mat-snackbar.service';
+import {
+  AuthCustomerService,
+  CustomerRegestrationForm,
+} from '../../services/customer-auth.service';
+import { CustomerResponse } from '../../services/interfaces';
+import { AuthService } from '../../services/auth.service';
+
+@Component({
+  imports: [
+    MatFormFieldModule,
+    MatButtonModule,
+    MatInputModule,
+    MatIconModule,
+    MatSelectModule,
+    MatDatepickerModule,
+    MatSnackBarModule,
+    FormsModule,
+    ReactiveFormsModule,
+    RouterOutlet,
+    RouterLink,
+    RouterLinkActive,
+  ],
+  providers: [provideNativeDateAdapter()],
+  standalone: true,
+  selector: 'app-registration-page',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.scss'],
+})
+export class RegistrationComponent implements OnInit {
+  isAuthenticated = computed(() => {
+    return this.authService.isAuthenticated();
+  });
+
+  registrationForm: FormGroup;
+  submitInProcess = signal(false);
+  hidepassword = true;
+
+  countries = ['Poland', 'United States', 'Canada'];
+
+  constructor(
+    private authService: AuthService,
+    private router: Router,
+    private fb: FormBuilder,
+    private CustomValidators: CustomValidatorsService,
+    private FormatData: FormatDataService,
+    private snackbarService: SnackbarService,
+    private authCustomerService: AuthCustomerService,
+  ) {
+    this.registrationForm = this.fb.group(
+      {
+        email: ['', [Validators.required, Validators.email]],
+        password: [
+          '',
+          [Validators.required, this.CustomValidators.passwordValidator],
+        ],
+        confirmPassword: ['', [Validators.required]],
+        firstName: ['', [Validators.required, Validators.pattern('[a-zA-Z]+')]],
+        lastName: ['', [Validators.required, Validators.pattern('[a-zA-Z]+')]],
+        dateOfBirth: [
+          '',
+          [
+            Validators.required,
+            this.CustomValidators.dateOfBirthValidator.bind(this),
+          ],
+        ],
+        street: ['', [Validators.required]],
+        city: ['', [Validators.required, Validators.pattern('[a-zA-Z]+')]],
+        country: [
+          '',
+          [
+            Validators.required,
+            this.CustomValidators.countryValidator(this.countries),
+          ],
+        ],
+        postalCode: [
+          '',
+          [Validators.required, this.CustomValidators.postalCodeValidator],
+        ],
+      },
+      {
+        validators: [
+          this.CustomValidators.confirmPasswordValidator(
+            'password',
+            'confirmPassword',
+          ),
+        ],
+      },
+    );
+  }
+
+  ngOnInit(): void {
+    if (this.isAuthenticated()) {
+      this.router.navigate(['/']);
+    }
+  }
+
+  onSubmit() {
```

Comment:
This funtion is too big. Please try to defragmentation it with private funtions.  Also please remove unnecessary consoles. If you need them add comments why do you need / DOTO remove at end of develop